File: z-clients/javascript/src/models/adminRoleUpsertOut.ts
// this file is @generated

export interface AdminRoleUpsertOut {
    id: string;
    created: Date;
    updated: Date;
}

export const AdminRoleUpsertOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminRoleUpsertOut {
        return {
            id: object['id'],
            created: new Date(Number(object['created'])),
            updated: new Date(Number(object['updated'])),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminRoleUpsertOut): any {
        return {
            'id': self.id,
            'created': self.created.getTime(),
            'updated': self.updated.getTime(),
        };
    }
}

File: z-clients/rust/src/models/admin_role_upsert_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminRoleUpsertOut {
    pub id: String,

    pub created: u64,

    pub updated: u64,
}

impl AdminRoleUpsertOut {
    pub fn new(id: String, created: u64, updated: u64) -> Self {
        Self {
            id,
            created,
            updated,
        }
    }
}
